src/Membership/MembershipRequestHandler.php

```diff
@@ -52,10 +60,25 @@ public function handle(MembershipRequest $membershipRequest)
         $activationUrl = $this->generateMembershipActivationUrl($adherent, $token);
         $this->mailer->sendMessage(AdherentAccountActivationMessage::createFromAdherent($adherent, $activationUrl));
 
-        $this->dispatcher->dispatch(AdherentEvents::REGISTRATION_COMPLETED, new AdherentAccountWasCreatedEvent($adherent, $membershipRequest));
         $this->dispatcher->dispatch(UserEvents::USER_CREATED, new UserEvent($adherent));
     }
 
+    public function join(Adherent $user, MembershipRequest $membershipRequest)
+    {
+        $user->updateMembership($membershipRequest, $this->addressFactory->createFromAddress($membershipRequest->getAddress()));
+        $user->join();
+        $this->manager->flush();
+
+        $this->mailer->sendMessage(AdherentAccountConfirmationMessage::createFromAdherent(
```

Comment:
Could you send the message with a subscriber to `AdherentAccountWasCreatedEvent` instead? It would avoid adding the managers as deps here.